backend/src/services/mirroredModel.ts

```diff
@@ -0,0 +1,224 @@
+    metadata: { modelId, exportKind: ExportKind.ModelCards },
+  })
+  // unknown release semver
+  if (releaseSemvers && releaseSemvers.length > 0) {
+    zipFiles.push({
+      filename: `${modelId}/releases.zip`,
+      file: await generateReleaseZip(user, model, releaseSemvers),
+      metadata: { modelId, exportKind: ExportKind.Releases },
+    })
+  }
+
+  return Promise.all(zipFiles.map((zipFile) => uploadZipFileToS3(zipFile)))
+}
+
+async function uploadZipFileToS3(zip: ZipFile) {
+  const s3Stream = new PassThrough()
+  const hashStream = new PassThrough()
+
+  zip.file.pipe(hashStream)
+  zip.file.pipe(s3Stream)
+
+  let signatures = {}
+  if (config.modelMirror.export.kmsSignature.enabled) {
+    try {
+      const messageDigest = await generateDigest(hashStream)
+      signatures = await sign(messageDigest)
+    } catch (error) {
+      throw InternalError('Failed to create signature for zip file.', {
+        error,
+        modelId: zip.metadata.modelId,
+        exportKind: zip.metadata.exportKind,
+      })
+    }
+  }
+
+  log.debug('Starting export of zip file to S3.', {
+    bucket: config.modelMirror.export.bucket,
+    name: zip.filename,
+    modelId: zip.metadata.modelId,
+    exportKind: zip.metadata.exportKind,
+  })
+  putObjectStream(config.modelMirror.export.bucket, zip.filename, s3Stream, signatures)
+    .then(() =>
+      log.debug('Successfully exported zip file to S3.', {
+        bucket: config.modelMirror.export.bucket,
+        name: zip.filename,
+        modelId: zip.metadata.modelId,
+        exportKind: zip.metadata.exportKind,
+      }),
+    )
+    .catch((error) =>
+      log.error('Failed to export zip file to S3.', {
+        error,
+        bucket: config.modelMirror.export.bucket,
+        name: zip.filename,
+        modelId: zip.metadata.modelId,
+        exportKind: zip.metadata.exportKind,
+      }),
+    )
+}
+
+async function generateModelCardRevisionsZip(user: UserInterface, model: ModelDoc) {
+  const cards = await getModelCardRevisions(user, model.id)
+
+  let zip: archiver.Archiver
+  try {
+    zip = archiver('zip')
+    for (const card of cards) {
+      zip.append(JSON.stringify(card.toJSON()), { name: `${card.version}.json` })
+    }
+    await zip.finalize()
+  } catch (error: any) {
+    throw InternalError('Error when generating the model card revisions zip file.', { error })
+  }
+  return zip
+}
+
+async function generateReleaseZip(user: UserInterface, model: ModelDoc, semvers: string[]) {
+  const releases = await getReleasesForExport(user, model.id, semvers)
+  if (semvers.length > 1) {
+    await checkTotalFileSize(model.id, semvers)
+  }
+
+  const zip = archiver('zip')
+  const errors: any[] = []
+  // Using a .catch here to ensure all errors are returned, rather than just the first error.
+  await Promise.all(releases.map((release) => addReleaseToZip(user, model, release, zip).catch((e) => errors.push(e))))
+  await zip.finalize()
```

Comment:
Archiver isn't an 'async / await' library.  The 'await' doesn't do anything here.  Also, I'm fairly confident you have to listen to events from the 'zip' created prior to finalizing or they could return early.